web/modules/custom/rdf_entity/src/Plugin/Field/FieldWidget/FileSizeCalculatorWidget.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+namespace Drupal\rdf_entity\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Entity\EntityFieldManagerInterface;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\file\Entity\File;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Plugin implementation of the 'file_size_calculator_widget' widget.
```

Comment:
Same as the other widget, except from being there for possible re-usability, the main idea is that this is to calculate an the file size of files in an entity and put this number to a field. But the field that holds the size is an rdf entity field and the file field is not necessarily. I could even move it to the joinup profile but it was created to solve problems for an rdf entity field so I thought to put it there.